File: services/intnet-admin/src/main/java/com/intnet/admin/features/intnetservice/model/IntnetService.java
package com.intnet.admin.features.intnetservice.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class IntnetService {
    private String name;
    private String helmChartPath;
    private String imageName;
    private String dockerfilePath;
    private ServiceType serviceType;
    private ServiceBuildType serviceBuildType;

    private ServiceKubernetesData kubernetesData;
}

File: frontends/admin-frontend/src/app/features/intnet-service/models/IntnetService.ts

export interface IntnetService {
    name: string;
    helmChartPath?: string;
    imageName?: string;
    dockerfilePath?: string;
    serviceType?: ServiceType;
    serviceBuildType?: ServiceBuildType;
    kubernetesData?: ServiceKubernetesData;
}

export interface ServiceKubernetesData {
    status: ServiceStatus;
    replicas: number;
    availableReplicas: number;
    namespace: string;
}

export enum ServiceType {
    CUSTOM = "CUSTOM",
    EXTERNAL = "EXTERNAL"
}

export enum ServiceBuildType {
    JAVA = "JAVA",
    PYTHON = "PYTHON",
    GO = "GO"
}

export enum ServiceStatus {
    RUNNING = "RUNNING",
    STOPPED = "STOPPED",
    PENDING = "PENDING",
    UNKNOWN = "UNKNOWN",
    ERROR = "ERROR"
}